File: app/src/main/java/com/wojdor/hearthcards/application/classcards/ClassCardsViewModel.java
package com.wojdor.hearthcards.application.classcards;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.wojdor.hearthcards.application.base.BaseAndroidViewModel;
import com.wojdor.hearthcards.domain.Card;

import java.util.List;

public class ClassCardsViewModel extends BaseAndroidViewModel {

    public ClassCardsViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<List<Card>> getCardsFromClass(String className) {
        return cardDao.getCardsFromClass(className);
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/app/classcards/ClassCardsViewModel.kt
package com.wojdor.hearthstonecards.app.classcards

import android.arch.lifecycle.LiveData
import com.wojdor.hearthstonecards.app.base.BaseViewModel
import com.wojdor.hearthstonecards.data.repository.CardRepository
import com.wojdor.hearthstonecards.domain.Card

class ClassCardsViewModel(cardRepository: CardRepository) : BaseViewModel(cardRepository) {

    fun getCardsFromClass(className: String): LiveData<List<Card>> = repository.getCardsFromClass(className)
}
